bpf/lib/nodeport.h

```diff
@@ -2907,6 +2914,22 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 	if (backend_local || !nodeport_uses_dsr4(&tuple)) {
 		struct ct_state ct_state = {};
 
+#if !(defined(ENABLE_CLUSTER_AWARE_ADDRESSING) && defined(ENABLE_INTER_CLUSTER_SNAT))
+		src_sec_identity = WORLD_IPV4_ID;
+#else
+		if (src_sec_identity == 0)
+			src_sec_identity = WORLD_IPV4_ID;
+#endif
+
+		 /* Before forwarding the identity, make sure it's not a CIDR
+		  * identity, as these are __u32 values, but transporting them
+		  * via the VNI field in the VXLAN / Geneve header allows for
+		  * only 24 bits.
+		  */
+
+		if (identity_is_cidr_range(src_sec_identity))
```

Comment:
I wonder also whether we shouldn't just be checking "in CIDR range", but actually the full upper 8 bits check. Comparing this with `pkg/identity/numericidentity.go` "scopes" I also see the remote node scope, which could use a range from `2<<24` up , which would be outside the range that this function checks :thinking: